multichain-testing/test/xcs-swap-anything/helpers.ts

```diff
@@ -0,0 +1,826 @@
+      {
+        typeUrl: '/cosmwasm.wasm.v1.MsgExecuteContract',
+        value: message,
+      },
+    ];
+
+    const response = await osmosisClient.signAndBroadcast(
+      osmosisAddress,
+      encodeObjects,
+      fee,
+    );
+    console.log('invokeOsmosisContract DeliverTxResponse: ', response);
+
+    const { msgResponses, code } = response;
+
+    if (code !== 0) {
+      throw Error(`Failed to execute osmosis contract with message ${message}`);
+    }
+
+    return msgResponses;
+  };
+
+  const queryOsmosisContract = async (contractAddress: string, queryMsg) => {
+    const wasmClient = await CosmWasmClient.connect(osmosisRpcEndpoint);
+    const result = await wasmClient.queryContractSmart(
+      contractAddress,
+      queryMsg,
+    );
+
+    return result;
+  };
+
+  const downloadXcsContracts = async (
+    xcsContracts: Contracts,
+    branch: string = osmosisBranch,
+    scriptPath: string = scriptLocalPath,
+    artifactsPath: string = xcsArtifactsPath,
+  ) => {
+    const wasmFiles: string[] = [];
+    for (const contract in xcsContracts) {
+      if (Object.prototype.hasOwnProperty.call(xcsContracts, contract)) {
+        const { label } = xcsContracts[contract];
+        wasmFiles.push(`${label}.wasm`);
+      }
+    }
+
+    try {
+      await execa('bash', [
+        scriptPath,
+        'osmosis-labs',
+        'osmosis',
+        branch,
+        'tests/ibc-hooks/bytecode',
+        artifactsPath,
+        ...wasmFiles,
+      ]);
+    } catch (error) {
+      throw Error(`Failed to download XCS artifacts: ${error}`);
+    }
+  };
+
+  const instantiateSingleXcsContract = async (
+    contractLabel: string,
+    instantiateArgs: CommonArgs | CrosschainSwapsArgs,
+  ) => {
+    const clientRegistry = osmosisClient.registry;
+    clientRegistry.register(MsgStoreCode.typeUrl, MsgStoreCode);
+
+    clientRegistry.register(
+      MsgInstantiateContract.typeUrl,
+      MsgInstantiateContract,
+    );
+
+    const contractPath = `${xcsArtifactsPath}/${contractLabel}.wasm`;
+
+    const wasmPath = path.resolve(contractPath);
+    const wasmByteCode = await fs.readFile(wasmPath);
+
+    const fee = {
+      amount: [{ denom: 'uosmo', amount: '100000' }],
+      gas: '10000000',
+    };
+
+    const storeMessage = MsgStoreCode.fromPartial({
+      sender: osmosisAddress,
+      wasmByteCode,
+    });
+
+    const storeEncodeObjects: Array<
+      import('@cosmjs/proto-signing').EncodeObject
+    > = [
+      {
+        typeUrl: MsgStoreCode.typeUrl,
+        value: storeMessage,
+      },
+    ];
+
+    const storeResult = await osmosisClient.signAndBroadcast(
+      osmosisAddress,
+      storeEncodeObjects,
+      fee,
+    );
+
+    if (storeResult.msgResponses[0] && storeResult.code !== 0) {
+      throw Error(`Failed to store ${contractLabel} contract`);
+    }
+
+    const codeId = MsgStoreCodeResponse.decode(
+      storeResult.msgResponses[0].value,
+    ).codeId;
```

Comment:
to avoid some redundancy:

```suggestion
    const { codeId } = MsgStoreCodeResponse.decode(
      storeResult.msgResponses[0].value,
    );
```

Reply:
Change request implemented on commit: 9bdf4e3ca6d289d459fa0b7d514776e6dfe20231